docs/architecture.md

```diff
@@ -0,0 +1,133 @@
+    - [Scheduler][gh-kube-scheduler]
+    - [Proxy][gh-kube-proxy]
+- [etcd][io-etcd]
+- [runc][gh-runc]
+- [containerd][io-containerd]/[cri][gh-cri-api]
+- [CNI][gh-cni]: Canal a.k.a. [Calico][org-projectcalico] &amp; [Flannel][gh-flannel]
+- [CoreDNS][io-coredns]
+- [NGINX Ingress Controller][io-ingress-nginx]
+- [Metrics Server][gh-metrics-server]
+- [Helm][sh-helm]
+
+### Process Lifecycle
+
+#### Content Bootstrap
+
+RKE2 sources binaries and manifests to run both _server_ and _agent_ nodes from the RKE2 Runtime image.
+This means RKE2 scans `/var/lib/rancher/rke2/agent/images/*.tar` for the [`rancher/rke2-runtime`](https://hub.docker.com/r/rancher/rke2-runtime/tags) 
+image (with a tag correlating to the output of `rke2 --version`) by default and if it cannot be found, attempts to pull
+it from the network (a.k.a. Docker Hub). RKE2 then extracts `/bin/` from the image, flattening it into
+`/var/lib/rancher/rke2/data/${RKE2_DATA_KEY}/bin` where `${RKE2_DATA_KEY}` represents a unique string identifying the
+image. RKE2 then extracts `/charts/` from the image into the `/var/lib/rancher/rke2/server/manifests` directory.
+
+#### Initialize Server
+
+Servers are specialized agents which means that the following startup is deferred until the container runtime has
+successfully started.
+
+##### Prepare Components
+
+###### `kube-apiserver`
+
+Pull the `kube-apiserver` image, if not present already, and spin up goroutine to lay down the static pod (after `etcd` is ready).
+
+###### `kube-controller-manager`
+
+Pull the `kube-controller-manager` image, if not present already, and spin up goroutine to lay down the static pod (after `kube-apiserver` is ready).
+
+###### `kube-scheduler`
+
+Pull the `kube-scheduler` image, if not present already, and spin up goroutine to lay down the static pod (after `kube-apiserver` is ready).
+
+##### Start Cluster
+
+Spin up an HTTP server in a goroutine to listen for other cluster servers then initialize/join the cluster. 
+
+###### `etcd`
+
+Pull the `etcd` image, if not present already, and spin up goroutine to lay down the static pod (after `kubelet` is ready).
+
+###### `helm-controller`
+
+Spin up the goroutine to start the embedded helm-controller (after `kube-apiserver` is ready).
+
+#### Initialize Agent
+
+These actions occur for both server and agent nodes.
+
+##### Container Runtime
+
+###### `containerd`
+
+Spawn the `containerd` process and listen for termination. If `containerd` exits then the `rke2` process will also exit.
+
+##### Node Agent
+
+###### `kubelet`
+
+Spawn and supervise the `kubelet` process. If `kubelet` exits then `rke2` will attempt to restart it.
+Once the `kubelet` is running the embedded K3s engine it will start any available static pods.
+For servers this means that `etcd` and `kube-apiserver` will start, in succession, allowing the remaining components
+started via static pod to connect to the `kube-apiserver` and begin their processing. 
+
+##### Server Charts
```

Comment:
Would it be worth a quick mention here that these server charts can be disabled via the `--disable` flag?  For those that want to bring their own ingress for example.

Reply:
I really want this to be an architecture overview. We'll have separate installation/configuraiotn instructions. I'd prefer that be there, to keep this tight.

Reply:
That sounds good to me 👍